docs/MAINTAINERS.adoc

```diff
@@ -0,0 +1,162 @@
+They are also responsible to monitor the weekly release status and to perform triage of critical issues.
+GitHub team: link:TODO[@jenkinsci/core].
+
+**Release Team Members** are responsible for Jenkins weekly and LTS releases.
+Led by the link:TODO[Jenkins Release Officer], they initiate releases, prepare changelogs and backport changes into the link:TODO[Stable branch].
+Team members get `Admin` permissions in the Jenkins core repository, and they also get permissions to trigger release Pipelines.
+
+=== Ladder
+
+* Anyone can participate as a **committer** or as a **reviewer**.
+  There is no precondition except having a GitHub account, just submit pull requests or comment in existing ones!
+* **Core Pull Request Reviewers** can be nominated by contributors in the link:https://groups.google.com/d/forum/jenkinsci-dev[developer mailing list]. 
+Self-nomination are fine.
+Decision is made by a consensus in the mailing list or via voting at the link:https://jenkins.io/project/governance-meeting/[governance meeting].
+* **Core Maintainers** - same process as for Core PR reviewers.
+  All nominees must also sign an link:https://github.com/jenkinsci/infra-cla/[Individual Contributor License Agreement] before getting the permission in GitHub repositories.
+* **Release Team Members** are assigned by the Jenkins Release officer
+
+== Pull request review process
+
+Jenkins core is a mission-critical part of the ecosystem.
+We need to ensure that submitted pull requests are not only code complete,
+but also that they do not introduce undesired defects, breaking changes and technical debt.
+At the same time, we are interested to make the review process as simple as possible for contributors and maintainers.
+
+=== Code review goals
+
+* Ensure that the pull request is feasible.
+** Not every change needs to be merged into the core, many features would be rather introduced as plugins.
+   Separate release cycles for plugins also help to deliver changes faster.
+** We want to extend the Jenkins core and incorporate widely used functionality there,
+   but we try to keep the core as minimal as possible in terms of dependencies.
+* Ensure that there is a consensus about the suggested change.
+** Not all changes are discussed before they submitted as pull requests.
+   Developer mailing lists, Jira issues and JEPs are used for discussions,
+   but sometimes the changes go straight to the pull requests.
+   And we are fine with that, especially for small patches.
+** Pull requests often become a venue to discuss feasibility, underlying technical decisions and design.
+   We are fine with that as well
+** If there is no consensus about the feasibility and implementation,
+   code reviewers are expected to suggest proper channels to maintainers
+* Ensure that the pull request matches the code quality practices: test automation, Javadoc for newly introduced APIs, etc.
+** There is no goal to make the code perfect tho
```

Comment:
API additions for use in plugins need to be carefully considered, however. When in doubt, `@Beta` or `@NoExternalUse` to make them effectively not _real_ API additions.